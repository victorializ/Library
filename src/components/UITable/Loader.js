import React from 'react';
import { Loader as SemanticLoader } from 'semantic-ui-react';

function Loader({ 
    loading = true
}) {
    return (
        loading && 
            <div className='center'>
                <SemanticLoader active inline='centered'/>
            </div>
    );
}

export { Loader };