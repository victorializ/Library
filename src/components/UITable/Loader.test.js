import React from 'react';
import { shallow } from 'enzyme';

import { Loader } from './Loader';

describe('Loader component', () => {
    it('renders correctly', () => {
        expect(shallow(<Loader />)).toMatchSnapshot();
    });   
    it('displays loader by default', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper.find('div').exists()).toBeTruthy();
    }); 
    it('displays loader when loading', () => {
        const wrapper = shallow(<Loader loading={true} />);
        expect(wrapper.find('div').exists()).toBeTruthy();
    });
    it('not displays loader when not loading', () => {
        const wrapper = shallow(<Loader loading={false} />);
        expect(wrapper.find('div').exists()).toBeFalsy();
    });
});