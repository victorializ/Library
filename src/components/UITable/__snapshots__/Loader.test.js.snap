// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders correctly 1`] = `
<div
  className="center"
>
  <Loader
    active={true}
    inline="centered"
  />
</div>
`;
